src/ui/storyTellingServerUI.js

```diff
@@ -0,0 +1,143 @@
+        }).fail(function (error) {
+            storyPromise.reject(error);
+        });;
+    } else {
+        storyPromise.reject();
+    }
+
+    return storyPromise;
+};
+
+/* Loads a story Browse page and populates it with a set of stories
+ * - "theme": the theme of the story Browse page (pass "page" in for base theme)
+ * - "options": additional options to merge into the Browse page
+ */
+sjrk.storyTelling.loadBrowse = function (theme, options) {
+    var storiesPromise = fluid.promise();
+
+    $.get("/stories", function (data) {
+        var browseResponse = JSON.parse(data);
+
+        options = options || {};
+        options.distributeOptions = {
+            "target": "{that storyBrowser}.options.model",
+            "record": browseResponse
+        };
+
+        var storyBrowseComponent = sjrk.storyTelling[theme].storyBrowse(options);
+        storiesPromise.resolve(storyBrowseComponent);
+    }).fail(function (error) {
+        storiesPromise.reject(error);
+    });
+
+    return storiesPromise;
+};
+
+/* Loads custom theme files if and only if one of either the theme override or
+ * server configuration theme are present AND not set to the base theme.
+ * Once complete, the provided callback function is called.
+ * - "callback": a function to call once everything has completed. Will be called
+ *               regardless of whether theme information was specified or retrieved
+ * - "themeOverride": allows overriding of the theme stored in the configuration
+ */
+sjrk.storyTelling.loadThemedPage = function (callback, themeOverride) {
+    var loadPromise = fluid.promise();
+
+    var callbackFunction = typeof callback === "function" ? callback : fluid.getGlobalValue(callback);
+
+    $.get("/clientConfig").then(function (data) {
+        var theme = themeOverride ? themeOverride : data.clientConfig ? data.clientConfig.theme : "base";
+
+        if (theme && theme !== "base") {
+            return sjrk.storyTelling.loadCustomThemeFiles(callbackFunction, theme).then(function () {
+                loadPromise.resolve(theme);
+            }, function (error) {
+                loadPromise.reject(error);
+            });
+        } else {
+            callbackFunction("page");
+            loadPromise.resolve(theme);
+        }
+    }, function () {
+        callbackFunction("page");
+        loadPromise.reject();
+    });
+
+    return loadPromise;
+};
+
+/* Loads CSS and JavaScript files for the provided theme into the page markup.
+ * If JavaScript file loading is successful, the callback function is called.
+ * - "theme": the theme of the story Browse page (pass "page" in for base theme)
+ * - "callback": a function to call once everything has completed
+ */
+sjrk.storyTelling.loadCustomThemeFiles = function (callback, theme) {
+    var loadPromise = fluid.promise();
+
+    var cssUrl = fluid.stringTemplate("/css/%theme.css", {theme: theme});
+    var scriptUrl = fluid.stringTemplate("/js/%theme.js", {theme: theme});
```

Comment:
It seems limited to only allow users to write one js file and one css file for their custom themes. Users would have more freedom if these type of files can be included in their custom templates such as a header template.

Since this approach satisfies the requirement for all two custom themes we have so far, let's stay with allowing only one for each. It can be improved when the requirement goes beyond its limit.

Reply:
I agree, and I've filed [SJRK-256](https://issues.fluidproject.org/browse/SJRK-256) to keep track of this suggestion. :)